fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/DefaultPredicateSelectivityEstimator.java

```diff
@@ -0,0 +1,371 @@
+import com.starrocks.sql.optimizer.ConstantOperatorUtils;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperatorVisitor;
+import com.starrocks.sql.optimizer.statistics.ColumnStatistic;
+import com.starrocks.sql.optimizer.statistics.ExpressionStatisticCalculator;
+import com.starrocks.sql.optimizer.statistics.PredicateStatisticsCalculator;
+import com.starrocks.sql.optimizer.statistics.Statistics;
+
+import java.util.Optional;
+
+/**
+ * Estimate Predicate selectivity which value between 0.0 and 1.0
+ * mixed PredicateStatisticsCalculator & Hive FilterSelectivityEstimator.java & Spark FilterEstimation.scala
+ */
+public class DefaultPredicateSelectivityEstimator {
+
+    private static final double SELECTIVITY_MIN = 0.0;
+
+    private static final double SELECTIVITY_MAX = 1.0;
+    //"Database Systems, the complete book". Suggest this:
+    private static final double SELECTIVITY_EQUALS = 1.0 / 3.0;
+
+    /**
+     * Returns a percentage of predicate selectivity
+     * If it's a single predicate, we estimate the percentage directly.
+     * If it's a compound predicate, it is decomposed into multiple single predicate linked with
+     * AND, OR, NOT.
+     *
+     * @param predicate  the predicate which need to estimate
+     * @param statistics the statistics which in OptExpression
+     * @return an optional double value to get the percentage of predicate selectivity
+     * It returns 1.0 if the condition is not supported.
+     */
+    public double estimate(ScalarOperator predicate, Statistics statistics) {
+        // if Optional is null, return SELECTIVITY_MAX by default.
+        return predicate.accept(new PredicateSelectivityEstimatorVisitor(statistics), null).orElse(SELECTIVITY_MAX);
+    }
+
+    private static class PredicateSelectivityEstimatorVisitor extends ScalarOperatorVisitor<Optional<Double>, Void> {
+
+        private final Statistics statistics;
+
+        public PredicateSelectivityEstimatorVisitor(Statistics statistics) {
+            this.statistics = statistics;
+        }
+
+        @Override
+        public Optional<Double> visit(ScalarOperator scalarOperator, Void context) {
+            //follow StatisticsCalculator line 1242 logic default
+            Statistics estimatedStatistics = PredicateStatisticsCalculator.statisticsCalculate(scalarOperator, statistics);
+            return Optional.of(estimatedStatistics.getOutputRowCount() / statistics.getOutputRowCount());
+        }
+
+        @Override
+        public Optional<Double> visitBinaryPredicate(BinaryPredicateOperator predicate, Void context) {
+            Preconditions.checkState(predicate.getChildren().size() == 2);
```

Comment:
Column statisitics could be UNKNOWN type, you should deal with the situation

Reply:
If column statisitics  is type of UNKNOWN, return 1.0 now.